test/modifiers/batch.test.js

```diff
@@ -0,0 +1,284 @@
+        .on('data', function (chunk) {
+          chunks += 1;
+          expect(chunk).to.be.an('array').and.to.have.lengthOf(COUNT);
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunks).to.equal(CHUNKS);
+          done();
+        });
+    });
+  });
+
+  describe('when both "time" and "count" are defined in options', function () {
+    function asyncWrite(stream, dataArr) {
+      if (dataArr.length) {
+        setTimeout(function () {
+          stream.push(dataArr[0]);
+
+          asyncWrite(stream, dataArr.slice(1));
+        }, 1);
+      } else {
+        stream.end();
+      }
+    }
+
+    it('writes chunks if count is met first before the defined time', function (done) {
+      var input = through.obj();
+      var chunks = 0;
+      var CHUNKS = 10;
+
+      input
+        .pipe(batch({
+          count: 1,
+          time: 5000
+        }))
+        .on('data', function (chunk) {
+          chunks += 1;
+          expect(chunk).to.be.an('array').and.to.have.lengthOf(1);
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunks).to.equal(CHUNKS);
+          done();
+        });
+
+      asyncWrite(input, _.times(CHUNKS));
+    });
+
+    it('writes chunks if time is met first before the defined count', function (done) {
+      var input = through.obj();
+      var CHUNKS = 10;
+      var reads = 0;
+
+      input
+        .pipe(batch({
+          count: 100,
+          time: 1
+        }))
+        .on('data', function (chunk) {
+          reads += 1;
+          expect(chunk).to.be.an('array').and.to.have.length.above(0);
+        })
+        .on('error', done)
+        .on('end', function () {
+          // for good measure, leave some wiggle room as to
+          // how many writes we expect, because timers
+          expect(reads).to.be.above(CHUNKS / 2);
+          done();
+        });
+
+      asyncWrite(input, _.times(10));
+    });
+
+    it('successfully ends if there is no data written', function (done) {
+      var input = getReadableStream([], {
+        objectMode: true
+      });
+
+      var chunksRead = false;
+
+      input
+        .pipe(batch({
+          count: 2,
+          time: 1
+        }))
+        .on('data', function (chunk) {
+          chunksRead = true;
+          expect(chunk).to.be.an('array');
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunksRead).to.equal(false);
+          done();
+        });
+    });
+  });
+
+  describe('when neither "time" nor "count" are defined in options', function () {
+    var expected = data.map(function (val) {
+      return [buff(val)];
+    });
+    var objExpected = objData.map(function (val) {
+      return [val];
+    });
+
+    basicTests(null, expected, objExpected);
+
+    it('writes chunks immediately when read', function (done) {
+      var DATA = _.times(5);
+      var input = through.obj();
```

Comment:
Can you use the `getDuplexStream` helper instead? `through` uses readable-stream which normalize the stream implementation. the `getDuplexStream` and `getReadableStream` helpers use whatever stream implementation came with the version of node we are testing on. So, we can claim compatibility since we actually use those streams.